src/components/tabs/js/tabDirective.js

```diff
@@ -114,6 +114,14 @@ function MdTab () {
         }
     );
     scope.$on('$destroy', function () { ctrl.removeTab(data); });
+  }
 
+  function firstChild (element, tagName) {
+    var children = element[0].children;
+    for (var i = 0; i < children.length; i++) {
```

Comment:
When you check the `length` property of an `HTMLCollection` object, it re-checks the DOM, which is slow. You should access the `length` only once.